tests/integration/proxy/testproxymain.cpp

```diff
@@ -511,6 +511,28 @@ void TestProxyImpl::on_test_signal_boolean_cb(const bool s) {
     record_signal();
 }
 
+void TestProxyImpl::on_notification_received()
+{
+    m_pending_notifications--;
+    if (m_pending_notifications == 0) {
```

Comment:
Unnecessary if statement... just decrement counter and call `check_done()`.. that checks that all counters are 0? If you want to do this change the same can be done in `record_signal()`.

Reply:
Silly me, indeed :-)